### ISSUE TITLE
The Llama 3 Herd of Models (vision part only)

### ISSUE THREAD

**Issue by JihoonJ** [paperswithlove/papers-we-read#41]
![image](https://github.com/user-attachments/assets/468d08ff-599c-4877-bb2c-4fc2482ae907)

## Links

1. Tech Report: https://ai.meta.com/research/publications/the-llama-3-herd-of-models/
2. Model: https://huggingface.co/collections/meta-llama/llama-31-669fc079a0c406a149a5738f 

## Insight
 - 아직 개발 중인 Multimodal Model의 Architecture와 Data, Training Recipe 등을 공개(not yet ready for release)
 - 지원하는 Multimodal은 Image, Video, speech로 이 중 visual part 위주로 review

## Architecture
![image](https://github.com/user-attachments/assets/6a7a83ea-c5fb-4c19-975f-cdbb955570d8)

1. Image Encoder
   - ViT-H/14 trained to align images and text with 2.5B image-text pairs for 5-epochs
   - pretraining
      - 224 x 224 (16 x 16 patches) with 7680 dim and 256 patches
      - fine-grained localization preserving을 위해 multi-layer feature extraction 사용(4th, 7th, 16th, 24th, 31th layer) + 8-gated self-attention layer (total 850M param = 630M + 220M)
      - The parameters of image encoder are not fronze during training stages.
2. Image Adapter
   - Flamingo와 같이 Visual token을 Cross Attention 형태로 LLM과 Adaption (every 4th self-attention layer)
   - Cross Attention 사용 시 Llama 3 450B 기준 100B 정도의 parameter 증가
   - Initial pre-training
      - 6B image-text pairs image 사용하였고, input image를 최대 336 x 336 tile 4개로 fitting(e.g., 672 x 672, 672 x 336, 1344 x 336)
   - Annealing
      - 500M annealing dataset으로 추가 학습.
      - 이 때 per-tile image resolutin을 증가하여 infographics understaning 같이 고해상도를 필요로 하는 task의 성능을 향상
3. Video Adapter
   - 최대 64 frame까지 입력 가능하고, frame image는 image encoder로 처리
   - 처리된 feature는 video aggregator에서 32개의 frame을 1개의 embed로 encoding 후 LLM cross attention으로 입력
   - video aggregator는 peceiver resampler를 사용
   - video aggregator와 cross attentin으로 Llama3 7B / 70B 기준 0.6B, 4.6B parameter 사용

## Performance
![image](https://github.com/user-attachments/assets/dc6909a8-6d32-4750-bd13-221a5dd3ba76)
Image Understanding Performance는 Llama 3-V 405B가 SoTA Closed API대비 월등하다고 할 수는 없지만, Llama 3-V 8B 기준 동급 open source SoTA model 대비 좋은 성능보임(MMMU 기준 Bunny-8B: 43.3)

![image](https://github.com/user-attachments/assets/dc00e734-dda9-4924-bb2e-74f936a585b8)
Video Understanding Performance는 비교 대상의 정량 평가가 많지 않아 상대적 비교는 어렵지만 괜찮은 성능을 보이는 것으로 보임
